### PR TITLE
Flag answers and questions implemented

### LINKED ISSUE #28: As an expert/user, I can flag answers and questions to prevent misinformation and spam.

### PR DESCRIPTION
**Pull request by dennycox**

### CODE REVIEW

File: Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+namespace AbracadabraAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FlagsController : ControllerBase
+    {
+        private readonly AbracadabraContext _context;
+        private readonly UserManager<ApplicationUser> userManager;
```
Comment (kevinbevers) [inline Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs]: _userManager is nicer if it is a readonly

**Review by kevinbevers — COMMENTED** [review]
looks good, implemented really fast.

File: Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+using AbracadabraAPI.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AbracadabraAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FlagsController : ControllerBase
```
Comment (VoidBGS) [inline Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs]: Maybe change the name to FlaggedEntitiesController*

File: Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs (hunk 1 above)
Comment (RobbeBryssinck) [reply Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/FlagsController.cs]: I agree we should keep it consistent, but the "_" is related to the variable being a member variable, not whether or not it's read only ;)

**Review by RobbeBryssinck — APPROVED** [review]

**Pull request merged**